src/02-common-principles.md

```diff
@@ -185,8 +186,96 @@ field in `dataset_description.json` of each subfolder of `derivatives` to:
 }
 ```
 
-See [Derived dataset and pipeline description](05-derivatives/01-introduction.md#derived-dataset-and-pipeline-description)
-for more information.
+### Storage of derived datasets
+
+Derivatives can be stored/distributed in two ways:
+
+1.  Under a `derivatives/` subfolder in the root of the source BIDS dataset
+    folder to make a clear distinction between raw data and results of data
+    processing. A data processing pipeline will typically have a dedicated directory
+    under which it stores all of its outputs. Different components of a pipeline can,
+    however, also be stored under different subfolders. There are few restrictions on
+    the directory names; it is RECOMMENDED to use the format `<pipeline>-<variant>` in
+    cases where it is anticipated that the same pipeline will output more than one variant (e.g.,
+    `AFNI-blurring`, `AFNI-noblurring`, etc.). For the sake of consistency, the
+    subfolder name SHOULD be the `GeneratedBy.Name` field in
+    `data_description.json`, optionally followed by a hyphen and a suffix (see
+    [Derived dataset and pipeline description][derived-dataset-description].
+
+    Example of derivatives with one directory per pipeline:
+
+    ```Plain
+    <dataset>/derivatives/fmriprep-v1.4.1/sub-0001
+    <dataset>/derivatives/spm/sub-0001
+    <dataset>/derivatives/vbm/sub-0001
+    ```
+
+    Example of a pipeline with split derivative directories:
+
+    ```Plain
+    <dataset>/derivatives/spm-preproc/sub-0001
+    <dataset>/derivatives/spm-stats/sub-0001
+    ```
+
+    Example of a pipeline with nested derivative directories:
+
+    ```Plain
+    <dataset>/derivatives/spm-preproc/sub-0001
+    <dataset>/derivatives/spm-preproc/derivatives/spm-stats/sub-0001
+    ```
+
+
+1.  As a standalone dataset independent of the source (raw or derived) BIDS
+    dataset.
+    This way of specifying derivatives is particularly useful when the source
+    dataset is provided with read-only access, for publishing derivatives as
+    independent bodies of work, or for describing derivatives that were created
+    from more than one source dataset.
+    The `sourcedata/` subdirectory MAY be used to include the source dataset(s)
+    that were used to generate the derivatives.
+    Likewise, any code used to generate the derivatives from the source data
+    MAY be included in the `code/` subdirectory.
+
+    Example of a derivative dataset including the raw dataset as source:
```

Comment:
Given the differential definitions of "source" vs. "raw" datasets in "common principles", it might be better to avoid use of the word "source" in the context of "data from which the derivatives were generated". That or "source" in reference to pre-BIDS-raw data would need to be changed.

There's also some potential ambiguity here in terms of hierarchical directory structure. It's permissible for derivative data to be stored in a sub-directory of the raw dataset, but it's also possible for the raw data to be stored in a sub-directory of the derivative dataset, and none of the directory names are prescribed; so it's a bit of a free-for-all, with the only restriction being storage of raw and derived data at the same hierarchical level. This would mean that for automatic parsing, unambiguously determining what's raw and what's derived would require running the entire validator on each candidate directory. Also, if the raw dataset is provided read-only, writing a read-write version of it along with the derivative data could be seen to be counteracting the intent of that read-only provision. If the raw data are to be in some way distributed alongside the derivative data, doing so according to the suggestion provided in "common principles":
```plain
my_dataset/
  sourcedata/
    ...
  rawdata/
    dataset_description.json
    participants.tsv
    sub-01/
    sub-02/
    ...
  derivatives/
    ...
```
may be more faithful to such.

Actually looking at it again, this is in "Common principles":
> These data MUST be kept in separate sourcedata and derivatives folders

Permitting BIDS-raw data to be stored as a sub-directory within a BIDS-derivatives directory could arguably be seen as contradictory to this requirement, since the derivatives directory is no longer "separate" to the (duplicate of the) raw data but a superset thereof.

So personally I'd probably err on *not* permitting raw data as sub-directories of derived data.

(Sorry I missed the timeline on this; I'll try to get through it nevertheless and you can use or not use as you wish)

Reply:
Potentially reinforcing this: consider #492. When reading data from the raw dataset, a parser will traverse up the filesystem hierarchy looking for JSON files that should be read in order. Unless the parser explicitly detects the root of the BIDS-raw directory as being such, and therefore ceasing traversal of the filesystem at that point, it will continue up and (potentially, depending on the design of the parser) include any JSON file at the root of the *derivatives* dataset in agglomerating sidecar information for a file in the *raw* dataset.